src/exec.rs

```diff
@@ -94,23 +94,30 @@ pub fn pacman<S: AsRef<str> + Display + std::fmt::Debug>(
     config: &Config,
     args: &Args<S>,
 ) -> Result<Status> {
-    let mut command = if config.need_root {
-        let mut command = Command::new(&config.sudo_bin);
-        command.args(&config.sudo_flags);
-        command.arg(args.bin.as_ref());
-        command
-    } else {
-        Command::new(args.bin.as_ref())
-    };
+    let mut command: std::process::Command;
+    let error_msg: String;
 
     if config.need_root {
         wait_for_lock(config);
+
+        command = Command::new(&config.sudo_bin);
+        command.args(&config.sudo_flags);
+        command.arg(args.bin.as_ref());
+        error_msg = format!("failed to run `{}` with arguments: {} {} {}",
```

Comment:
I don't see any need for the `with arguments` bit. Also this causes excess spaces when sudoflags is empty (which it usually will be).

Reply:
Right, I had thought about adding dealing with the extra space but I didn't feel like adding another 4 lines of code and an extra if/else statement just to deal with one space.

But you're right, just forget the arguments since this error will (I assume) only appear if the first program can't be spawned. Otherwise it will be up to that program to provide a suitable error message.

I will make the changes.